compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -734,13 +733,36 @@ abstract class GenJSCode extends plugins.PluginComponent
         if (scalaPrimitives.isPrimitive(sym)) {
           None
         } else if (sym.isDeferred || sym.owner.isInterface) {
+          val body = if (sym.hasAnnotation(JavaDefaultMethodAnnotation)) {
+            /* For an interface method with @JavaDefaultMethod, make it a
+             * default method calling the impl class method.
+             */
```

Comment:
So you are doing this because it is hard to recover the impl class method body at this point, right? This will not be necessary once scalac emits default methods, right?